backend/app/api/ws/device_namespace.py

```
@@ -673,26 +431,16 @@ async def on_disconnect(self, sid: str):
-                        # Emit task:status to notify frontend of task failure
-                        if info.task_id not in emitted_task_ids:
-                            await ws_emitter.emit_task_status(
-                                user_id=info.user_id,
-                                task_id=info.task_id,
-                                status="FAILED",
-                                progress=0,
-                            )
-                            emitted_task_ids.add(info.task_id)
+                extended_emitter = get_extended_emitter()
+                # Track unique task IDs to emit task:status only once per task
+                emitted_task_ids = set()
```

Ruff (0.15.0)

[error] Local variable `emitted_task_ids` is assigned to but never used

Remove assignment to unused variable `emitted_task_ids`

(F841)